src/exports.rs

```diff
@@ -23,7 +26,10 @@ pub extern "C" fn allocate(size: usize) -> *mut c_void {
 // It will free both the Slice and the memory referenced by the slice.
 #[no_mangle]
 pub extern "C" fn deallocate(pointer: *mut c_void) {
-    mem::drop(consume_slice(pointer));
+    let v = unsafe { consume_slice(pointer) };
+    if let Ok(buffer) = v {
+        mem::drop(buffer);
+    }
```

Comment:
If you just want to drop this, you can change this into 

```Rust
    let _ = unsafe { consume_slice(pointer) };
```

because `consume_slice` returns an owned type, so `drop` will be called when its lifetime ends, which in this case will be the end of the scope.

Reply:
Okay, I got so many compiler warnings about unused variables, I try to make sure to use them all now :) But will update.

So.. the `Result<Vec<u8>, Error>` will drop the Vec on success, and just ignore the error on failure. So much code disappearing. 😄